cpp/src/io/comp/uncomp.cpp

```diff
@@ -382,7 +382,7 @@ void io_uncompress_single_h2d(const void *src,
     int zerr = cpu_inflate_vector(dst, comp_data, comp_len);
     if (zerr != 0) {
       dst.resize(0);
-      CUDF_EXPECTS(0, "Decompression: error in stream");
+      CUDF_FAIL("Decompression: error in stream");
```

Comment:
Does resizing `dst` matter here (if an exception is being thrown)? I suppose this could be `CUDF_EXPECTS(zerr != 0, "Decompression: error in stream");`

Reply:
`dst` is an output parameter, not a local variable, so reseting it makes sense in theory. However, we should use a return value instead of an output parameter. So... stay tuned for more.

Reply:
Great, it's even better with the buffer as a return value.